lib/src/code_field.dart

```diff
@@ -263,7 +263,7 @@ class CodeFieldState extends State<CodeField> {
     );
 
     final numberCol = Container(
-      width: widget.lineNumberStyle.width,
+      width: (((_numberController?.text ?? '\n').split('\n')).last.toString().length) * widget.lineNumberStyle.width,
```

Comment:
This width calculation looks too complicated. Maybe, it is better to move the first part (before `* widget.lineNumber...` )  inside LineNumber controller (if possible)?

Reply:
It's not possible, because LineNumberController don't change after initialization.